From 164e798788bf7751ad8de7b1135f1ee590e18b57 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Mon, 11 Dec 2017 15:13:29 +0800
Subject: [PATCH 3075/5242] MLK-16715-9 ARM64: dts:
 fsl-imx8qm-lpddr4-arm2-hsic: add HSIC board dts

commit  b134dba29b8e342c2259c47cde18c76b745961f4 from
https://source.codeaurora.org/external/imx/linux-imx.git

Some special for HSIC usages:
- It needs imx8 debug board to test HSIC function.
- The USBOTG1 needs to be active when HSIC is in use.

BuildInfo:
- SCFW e0362348, IMX-MKIMAGE 9841373a, ATF e173337
- U-Boot 2017.03-imx_v2017.03+g3535868

Acked-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../dts/freescale/fsl-imx8qm-lpddr4-arm2-hsic.dts  |   75 ++++++++++++++++++++
 2 files changed, 77 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-hsic.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index cfd97b6..08d00e9 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -37,7 +37,8 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 				 fsl-imx8qm-lpddr4-arm2-dsi-adv7535.dtb \
 				 fsl-imx8qm-lpddr4-arm2-dsi-rm67191.dtb \
 				 fsl-imx8qm-lpddr4-arm2-it6263-adv7535.dtb \
-				 fsl-imx8qm-lpddr4-arm2-it6263-rm67191.dtb
+				 fsl-imx8qm-lpddr4-arm2-it6263-rm67191.dtb \
+				 fsl-imx8qm-lpddr4-arm2-hsic.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-mek.dtb \
 				  fsl-imx8qxp-mek-dom0.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-hsic.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-hsic.dts
new file mode 100644
index 0000000..d2ce4fe
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-hsic.dts
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/* This dts is only for verify USB HSIC function at i.MX8 Debug Board */
+
+#include "fsl-imx8qm-lpddr4-arm2.dts"
+
+/ {
+	imx8qm-pm {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		pd_usbh1: PD_USBH1 {
+			compatible = "nxp,imx8-pd";
+			reg = <SC_R_LAST>;
+			#power-domain-cells = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			pd_usbh1_io: usbh1_io {
+				reg = <SC_R_BOARD_R2>;
+				#power-domain-cells = <0>;
+				power-domains =<&pd_usbh1>;
+			};
+		};
+	};
+};
+
+&iomuxc {
+	imx8qm-arm2 {
+		pinctrl_usb_hsic_idle: usbh1_1 {
+			fsl,pins = <
+				SC_P_USB_HSIC0_DATA_CONN_USB_HSIC0_DATA		0xc60000c5
+				SC_P_USB_HSIC0_STROBE_CONN_USB_HSIC0_STROBE	0xc60000c5
+			>;
+		};
+
+		pinctrl_usb_hsic_active: usbh1_2 {
+			fsl,pins = <
+				SC_P_USB_HSIC0_STROBE_CONN_USB_HSIC0_STROBE	0xc60000d5
+			>;
+		};
+	};
+};
+
+&usbotg1 {
+	ci-disable-lpm;
+};
+
+/*
+ * Due to USB HSIC uses the same AHB and 480M with USBOTG1,
+ * the usbotg1 must be enabled for usbh1 usage.
+ */
+&usbh1 {
+	pinctrl-names = "idle", "active";
+	pinctrl-0 = <&pinctrl_usb_hsic_idle>;
+	pinctrl-1 = <&pinctrl_usb_hsic_active>;
+	power-domains = <&pd_usbh1_io>;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	disable-over-current;
+	status = "okay";
+};
-- 
1.7.9.5

